subworkflows/local/bismark.nf

```diff
@@ -13,21 +12,14 @@ include { BISMARK_SUMMARY                             } from '../../modules/nf-c
 
 workflow BISMARK {
     take:
-    reads  // channel: [ val(meta), [ reads ] ]
+    reads              // channel: [ val(meta), [ reads ] ]
+    bismark_index      // channel: /path/to/BismarkIndex/
+    skip_deduplication // boolean: whether to deduplicate alignments
+    cytosine_report    // boolean: whether the run coverage2cytosine
```

Comment:
Not sure if there's been discussion about this already, but I'm concerned that as parameters/switches proliferate, we'll end up with an unwieldy list of parameter input channels. Can we collapse this into a map? E.g. 
```groovy
take:
reads // [val(meta), [reads]]
bismark_index // /path/to/index
params // [skip_deduplication: true, cytosine_report: false]
```

Reply:
I'd prefer not to pass the entire params map. It is quite a bit simpler than that as many of the params "collapse" - for example, `params.skip_deduplication || params.rrbs` becomes `skip_deduplication`

I'd like to keep the inputs explicitly named so that anyone looking at this subworkflow can see what the inputs are without guessing / looking upstream at where it's called, this makes using it much easier. So I think a named tuple would be best if anything.

From there, I'm not sure if it makes all that much difference whether it's

```nextflow
take:
reads
bismark_index
tuple(
     skip_deduplication,
     cytosine_report
)
```

or 

```nextflow
take:
reads
bismark_index
skip_deduplication
cytosine_report
```

_(unfairly split the tuple over multiple lines for dramatic effect, but you get the idea)._

But I don't have a strong opinion either way.

Reply:
Note that command line flags don't come in here, these are likely only going to be switched to turn processes on and off. So I can't really foresee that list growing a huge amount in the future..

Reply:
yes, I see what you mean. Someone reusing the subworkflow would have to know exactly what keys and values to pass to the params map, which sounds like a bad plan. Putting them in channels makes it much clearer

Reply:
🤪 Dramatic tuple splitting FTW